### DISCUSSION TITLE
Automatic Speedtest

### DISCUSSION THREAD

**Discussion by tcoombs** [netalertx/NetAlertX#396] [Ideas]
Are there any plans for a plug-in to run an automatic speedtest? I believe this would be a useful additional feature for an already excellent solution.

**Comment by jokob-sk** [comment]
Hi @tcoombs ,

Have a look at the [plugin system](https://github.com/jokob-sk/Pi.Alert/blob/main/front/plugins/README.md) and you can submit a PR for it. it could be simply logging the average DL/UL speed. Later we could have a look at how to display Plugin entries as a chart, which I was already considering before. 

Regrettably, I have a full-time job and get up ~2h before my work starts just to maintain this project, so I have to be very selective about where I invest my time as this project doesn't bring me enough income to justify more time on it. Now, I'm focusing on making it more maintainable. 

Thanks,
j

**Reply by tcoombs** [reply]
Hi, will have a look at the plug-ins structure as suggested. I know that Okta Speedtest has a command line option and so this could be executed via a script. Cheers, Tony

**Comment by jokob-sk** [comment]
Implemented this and it can be tested in the `_dev` image
